### PR TITLE
adjusted recipes for default policy change

### PR DESCRIPTION
**Pull request by juliaElastic**
Fixes https://github.com/elastic/cloud-on-k8s/issues/5262
To adjust recipes to the change in kibana (removing default policies), added system, elastic_agent, fleet_server packages to the preconfigured policies for ECK.

### CODE REVIEW

**Comment by juliaElastic** [comment]
run full pr build

**Comment by juliaElastic** [comment]
run full pr build

**Comment by juliaElastic** [comment]
run full pr build

**Comment by juliaElastic** [comment]
run full pr build

**Comment by juliaElastic** [comment]
@naemono hi, I'm not sure what is wrong here, the build failed: https://devops-ci.elastic.co/blue/organizations/jenkins/cloud-on-k8s-pr/detail/cloud-on-k8s-pr/7918/pipeline

**Comment by pebrc** [comment]
@juliaElastic looks like we exceeded our disk quota in the GCP project the tests run in 

```
=Insufficient regional quota to satisfy request: resource "LOCAL_SSD_TOTAL_GB": request requires '1125.0' and is short '1125.0'. project has a quota of '30000.0' with '0.0' available
```

**Comment by juliaElastic** [comment]
hm, there are 2 failing e2e tests, is there a way to debug locally?
https://devops-ci.elastic.co/job/cloud-on-k8s-pr-full/107/#showFailuresLink

**Comment by juliaElastic** [comment]
run full pr build

**Review by joshdover — APPROVED** [review]
Changes LGTM from a Fleet perspective

**Comment by naemono** [comment]
@juliaElastic I'm going to try and replicate the failures you're seeing, and I'll get back to you about this.

File: docs/orchestrating-elastic-stack-applications/agent-fleet.asciidoc [hunk 1]
```diff
@@ -232,12 +264,48 @@ spec:
...
+              name: system
 ----
 
 *  `xpack.fleet.agents.elasticsearch.hosts`  must point to the Elasticsearch cluster that Elastic Agents should send data to. For ECK-managed Elasticsearch clusters, ECK creates a Service accessible through `https://ES_RESOURCE_NAME-es-http.ES_RESOURCE_NAMESPACE.svc:9200` URL, where `ES_RESOURCE_NAME` is the name of Elasticsearch resource and `ES_RESOURCE_NAMESPACE` is the namespace it was deployed in.
 
 *  `xpack.fleet.agents.fleet_server.hosts` must point to Fleet Server that Elastic Agents should connect to. For ECK-managed Fleet Server instances, ECK creates a Service accessible through `https://FS_RESOURCE_NAME-agent-http.FS_RESOURCE_NAMESPACE.svc:8220` URL, where `FS_RESOURCE_NAME` is the name of Elastic Agent resource with Fleet Server enabled and `FS_RESOURCE_NAMESPACE` is the namespace it was deployed in.
 
+*  `xpack.fleet.packages` are required packages to enable Fleet Server and Elastic Agents to enroll. 
+
+*  `xpack.fleet.agentPolicies` policies are needed for Fleet Server and Elastic Agents to enroll to, see https://www.elastic.co/guide/en/fleet/current/agent-policy.html for more information.
```
Comment (alaudazzi) [inline docs/orchestrating-elastic-stack-applications/agent-fleet.asciidoc:307]:
```suggestion
*  `xpack.fleet.agentPolicies` policies are needed for Fleet Server and Elastic Agents to enroll to, check https://www.elastic.co/guide/en/fleet/current/agent-policy.html for more information.
```
Avoiding "see" for accessibility purposes.
Ref [How to create accessible content](https://docs.google.com/document/d/1IGuMf16-cwxx8BX1Z9PWsFWsTPxicwTkX_r7UOhjd4E/edit#).

**Comment by naemono** [comment]
> @juliaElastic I'm going to try and replicate the failures you're seeing, and I'll get back to you about this.

I'm sorry, I ran into a large number of local issues yesterday trying to replicate this, I'm going to try again today.

**Comment by juliaElastic** [comment]
@naemono hi, any luck with finding out the root cause of this build failure?

**Comment by naemono** [comment]
Ok, I've finally had some luck replicating this, and I'll see if I can figure out the root cause. Sorry again for the delays.
```
=== RUN   TestFleetKubernetesIntegrationRecipe/ES_data_should_pass_validations
Retries (30m0s timeout): .............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
    step.go:43:
        	Error Trace:	utils.go:87
        	Error:      	Received unexpected error:
        	            	elasticsearch client failed for https://elasticsearch-ms5n-es-internal-http.e2e-2l2sg-mercury.svc:9200/_data_stream/logs-elastic_agent.filebeat-default: 404 Not Found: {Status:404 Error:{CausedBy:{Reason: Type:} Reason:no such index [logs-elastic_agent.filebeat-default] Type:index_not_found_exception RootCause:[{Reason:no such index [logs-elastic_agent.filebeat-default] Type:index_not_found_exception}]}}
        	Test:       	TestFleetKubernetesIntegrationRecipe/ES_data_should_pass_validations
{"log.level":"error","@timestamp":"2022-01-31T20:42:40.124Z","message":"stopping early","service.version":"0.0.0-SNAPSHOT+00000000","service.type":"eck","ecs.version":"1.4.0","error":"test failure","error.stack_trace":"github.com/elastic/cloud-on-k8s/test/e2e/test.StepList.RunSequential\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/step.go:44\ngithub.com/elastic/cloud-on-k8s/test/e2e/test/helper.RunFile\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/helper/yaml.go:162\ngithub.com/elastic/cloud-on-k8s/test/e2e/agent.runAgentRecipe\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/agent/recipes_test.go:226\ngithub.com/elastic/cloud-on-k8s/test/e2e/agent.TestFleetKubernetesIntegrationRecipe\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/agent/recipes_test.go:131\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:1259"}
```

**Comment by juliaElastic** [comment]
run full pr build

**Comment by juliaElastic** [comment]
I think the missing piece was `is_default` flag which was needed in` elastic-package` repo too: https://github.com/elastic/elastic-package/issues/655
Still waiting for the build to finish, hopefully it will pass.

**Comment by juliaElastic** [comment]
The build is passing, can someone approve to merge?

File: config/recipes/elastic-agent/fleet-kubernetes-integration.yaml [hunk 2]
```diff
@@ -11,20 +11,37 @@ spec:
...
+      version: latest
+    - name: elastic_agent
+      version: latest
+    - name: fleet_server
+      version: latest
     - name: kubernetes
       # pinning this version as the next one introduced a kube-proxy host setting default that breaks this recipe,
       # see https://github.com/elastic/integrations/pull/1565 for more details
-      version: 0.14.0
+      version: 0.14.0
```
Comment (thbkrkr) [inline config/recipes/elastic-agent/fleet-kubernetes-integration.yaml:23]:
```suggestion
      version: 0.14.0
```

File: config/recipes/elastic-agent/fleet-apm-integration.yaml [hunk 3]
```diff
@@ -32,7 +53,7 @@ spec:
           enabled: true
           vars:
           - name: host
-            value: 0.0.0.0:8200
+            value: 0.0.0.0:8200
```
Comment (thbkrkr) [inline config/recipes/elastic-agent/fleet-apm-integration.yaml:56]:
```suggestion
            value: 0.0.0.0:8200
```

File: config/recipes/elastic-agent/fleet-apm-integration.yaml [hunk 4]
```diff
@@ -11,19 +11,40 @@ spec:
...
           name: fleet_server
-        name: fleet_server-1
-    - name: Default Elastic Agent on ECK policy
+    - name: Elastic Agent on ECK policy
+      id: eck-agent
+      namespace: default
+      monitoring_enabled:
+      - logs
+      - metrics
+      unenroll_timeout: 900
```
Comment (thbkrkr) [inline config/recipes/elastic-agent/fleet-apm-integration.yaml:41]:
```suggestion
      unenroll_timeout: 900
```

**Review by thbkrkr — APPROVED** [review]
LGTM

Nit: trim trailing whitespaces.

**Comment by juliaElastic** [comment]
> LGTM
> 
> Nit: trim trailing whitespaces.

Thanks, committed your suggestions. Please merge when ready, I don't have access.

**Pull request merged**